### DISCUSSION TITLE
O que vamos fazer aqui

### DISCUSSION THREAD

**Discussion by ortense** [mist-gopher/feature-tag#1] [Request for Comments]
O objetivo aqui é ter um serviço para gestão de `feature tags` centralizado para múltiplos apps

## O que já sabemos sobre as funcionalidades desse serviço

- O serviço de feature tags deve permitir cadastrar um app
- O serviço de feature tags deve permitir cadastrar uma tag para um app
- O serviço de feature tags deve permitir atualizar uma tag para um app
- O serviço de feature tags deve permitir consultar as tags de um app

## O que já sabemos sobre as decisões técnicas desse serviço

- Vamos expor uma API HTTP para consumo do serviço
- Vamos usar [go 1.22](https://go.dev/dl/) (ultima versão disponível no momento da criação do repositório)
- Vamos utilizar um [redis](https://redis.io/) para armazenar as tags

![image](https://github.com/mist-gopher/feature-tag/assets/1520183/da62d601-c2a3-4368-bdd1-8a36a45aedca)

**Comment by tassiorego** [comment]
Pensando em reutilizar em diversos contexto devemos ser capaz de adicionar uma tag para um `user` específico, para pensar na ideia do rollout?

**Comment by mottajunior** [comment]
Sera que faz sentido, sempre que for solicitado para a API, obter tags de uma loja que ainda nao esteja no cache, o serviço vai buscar todas tags daquela loja para cachear?  No final do dia queremos ter um endpoint que responda a seguinte pergunta: **Esta loja, tem esta tag?**
